assignments/asgmt6.rb

```diff
@@ -0,0 +1,86 @@
+ require 'minitest/autorun'
+  
+  class PriorityQueue
+
+    class SubQueue
+      attr :rank, :items
+      def initialize(rank)
+         @rank = rank
+         @items = []
+      end
+      def enqueue(item)
+        @items<<item
+      end
+      def dequeue
+        @items.shift
+      end
+      def empty?
+        @items.empty?
+      end
+      def peek
+        @items[0]
+      end
+      def length
+        @items.length
+      end
+    end #SubQueue
+    
+    def initialize
+      @high_sq = SubQueue.new(:high)
+      @medium_sq = SubQueue.new(:medium)
+      @low_sq = SubQueue.new(:low)
+      @sq_names = {high: @high_sq, medium: @medium_sq, low: @low_sq} 
+    end
+    def enqueue(item, priority=:medium)
+      subqueue = @sq_names.select {|i| i == priority}.values[0]
```

Comment:
why not just use priority as the key for your `@sq_names` hash? Then you could write: `subqeue = @sq_names[priority]`